cargo/cargo_build_script.bzl

```diff
@@ -114,33 +123,46 @@ def _cargo_build_script_run(ctx, script):
         ),
     ]
 
-def _build_script_impl(ctx):
-    return _cargo_build_script_run(ctx, ctx.executable.script)
-
 _build_script_run = rule(
-    _build_script_impl,
+    doc = (
```

Comment:
Continuing https://github.com/bazelbuild/rules_rust/pull/419#discussion_r498754180

Reply:
@damienmg I think it's correct to document both the wrapper and the rule. If this were to be a bunch of macro's calling each other, I would agree that there should simply be a reference to whichever the top macro was.

Reply:
Can you extend your reasoning for duplicating the documentation in various place?

Reply:
Now I see the commit again, this document SGTM.

Reply:
1. If I'm reading a rule's implementation and see `ctx.attr.foo`. My first thought is "Hmm, what's `foo`? Let me go check the rule's attributes". And if all I see is `"foo": attr.string(),`, then I'll be pretty annoyed. Even if somewhere there's a macro that has documentation for `foo`, it's not where I would have looked to find this information and nothing told me where to look. 

2. I don't think it's ever acceptable to have undocumented parameters just because there's documentation for it somewhere else. Take the example below
```python
def thing(something, foo):
    #...
    return bar

def stuff(something, baz):
    #...
    return left, right

def place(something, zap):
    #...
    return some_zoom

def impl(ctx)

    something = ctx.attr.something if ctx.attr.maybe else ctx.attr.else

    cool = thing(something, "yay")

    ctx.actions.run(
        #...
        inputs = [cool]
    )

    # ...

    place(
        stuff(soemthing_new, cool),
        "nice",
    )

    return [DefaultInfo(files=cool)]
```

Even though `something` is a parameter of each macro, I should still be able to read the macro by itself and understand what it's inputs are and what it intends to do with them.  If it turns out `def thing(something, foo):` is something generic like 
```python
def thing(something, foo):
    return something + foo
```
Then this likely shouldn't exist or the signature should change to be more generic. However, if there truly is some reason why this signature should stay the same, it should be documented and each parameter (with it's type information) should be included in that documentation.